flask/app/buckets.py

```diff
@@ -17,54 +17,80 @@
 @login_required
 def get_unlinked_files():
 
-    minio_client = get_minio_client()
-
-    app.logger.debug("Getting all minio bucket names..")
-    all_bucket_names = [bucket.name for bucket in minio_client.list_buckets()]
-
-    # Remove buckets the current user does not have access to assuming group_code.lower() == bucket name
-    app.logger.debug("Getting all buckets that user has access to..")
-    user = (
-        models.User.query.filter_by(user_id=current_user.user_id)
-        .options(joinedload(models.User.groups))
-        .first()
-    )
-
-    valid_bucket_names = []
-    for group in user.groups:
-        code = group.group_code.lower()
-        if code in all_bucket_names:
-            valid_bucket_names.append(code)
-
-    app.logger.debug("Getting all files in valid minio buckets..")
-    all_files = []
-    for bucket in valid_bucket_names:
-        objs = minio_client.list_objects(bucket, recursive=True)
-        for obj in objs:
-            all_files.append(bucket + "/" + obj.object_name)
-
-    app.logger.debug("Getting all linked files..")
+    path = request.args.get("path", type=str)
```

Comment:
Please set a default value here or you'll error out on a plain `/api/unlinked`. Otherwise input validation doesn't have to be in scope of this PR since it wasn't a thing here originally.